src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -107,42 +107,29 @@ public TxReceipt[] Get(Block block)
                 return Array.Empty<TxReceipt>();
             }
 
-            if (_receiptsCache.TryGet(block.Hash, out var receipts))
+            return Get(block.Hash);
+        }
+        
+        public TxReceipt[] Get(Keccak blockHash)
+        {
+            if (_receiptsCache.TryGet(blockHash, out var receipts))
             {
-                return receipts;
+                return receipts ?? Array.Empty<TxReceipt>();
             }
             
-            var receiptsData = _blocksDb.GetSpan(block.Hash);
+            var receiptsData = _blocksDb.GetSpan(blockHash);
             try
             {
-                bool shouldCache = true;
-                
-                if (!receiptsData.IsNullOrEmpty())
+                if (receiptsData.IsNullOrEmpty())
                 {
-                    receipts = DecodeArray(receiptsData);
+                    return Array.Empty<TxReceipt>();
                 }
                 else
                 {
-                    // didn't bring performance uplift that was expected
-                    // var data = _database.MultiGet(block.Transactions.Select(t => t.Hash));
-                    // return data.Select(kvp => DeserializeObsolete(new Keccak(kvp.Key), kvp.Value)).ToArray();
-
-                    receipts = new TxReceipt[block.Transactions.Length];
-                    for (int i = 0; i < block.Transactions.Length; i++)
-                    {
-                        receipts[i] = FindReceiptObsolete(block.Transactions[i].Hash);
-                        shouldCache &= receipts[i] != null;
-                    }
-                }
-
-                shouldCache &= receipts.Length > 0;
```

Comment:
We potentially loose support for old database format where receipts were stored per transaction hash and not per block hash.

Question is it still needed. Format was changed 2 years ago and we do have database migration for it.

Reply:
We were using this logic in `Get(block)` and not using in `Get(blockHash)`. I wanted to make returns consistent. I can add it back, but now for both, `Get(block)` and `Get(blockHash)` (actually add to `Get(blockHash)` and reuse in `Get(block)`)

Reply:
Ahh, we actually cant add it to `Get(blockHash)`, because we don't have `block.Transactions` there. So question now, it is better to drop support for old db and have consistent returns, or revert changes?

Reply:
ok lets drop it for now